src/RosClientNode.hpp

```diff
@@ -39,12 +41,20 @@ class RosClientNode : public QObject {
   template <typename T>
   void encode_ros_msg(
       const T& msg, const std::string& msg_type, const std::string& to_topic) {
-    flatbuffers::FlatBufferBuilder fbb;
-    auto metadata = encode_metadata(fbb, msg_type, to_topic);
-    auto root_offset = encode<T>(fbb, msg, metadata);
-    fbb.Finish(flatbuffers::Offset<void>(root_offset));
-    Q_EMIT ros_message_encoded(QByteArray(
-        reinterpret_cast<const char*>(fbb.GetBufferPointer()), fbb.GetSize()));
+    auto& rate_info = rate_limits[to_topic];
+    auto curr_time = std::chrono::high_resolution_clock::now();
+    auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(curr_time - rate_info.first);
+    double elapsed = duration.count() / 1000.0;
+
+    if (elapsed > rate_info.second) {
```

Comment:
Note: if we get topics more frequently than the rate limit, we simply drop them.
Is this correct? Seems better than sending them late but open to other options

Reply:
Yes, this is the correct behavior: drop more frequent messages.